lib/firestore/models.dart

```diff
@@ -200,6 +255,140 @@ class Page<T> extends ListBase<T> {
   }
 }
 
+class Query extends Reference {
```

Comment:
Can you rename this class to `QueryReference` to match the other `Reference` implementations?